recaptcha_enterprise/demosite/app/static/scripts/global.js

```diff
@@ -12,47 +12,120 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// ATTENTION: reCAPTCHA Example (Part 3) Starts
-// SEE: https://cloud.google.com/recaptcha-enterprise/docs/create-assessment
-// SEE: If using a library or famework, can use event handlers its usual way
+const SUBMIT_URLS = {
+  comment: "on_comment_submit",
+  home: "on_homepage_load",
+  login: "on_login",
+  signup: "on_signup",
+  store: "on_store_checkout",
+  game: undefined,
+};
 
-function createAssessment({ action, token }) {
-  // SEE: Code for fetching the assessment from backend goes here
-  // SEE: Refer to demo app backend code for more information
-  // SEE: If using a library or framework, can fetch its usual way
-  return fetchDemoAssessment({ action, token });
+async function homepage({token}) {
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      token: token,
+    },
+  });
+  // Fetch the demo assessment from backend.
+  const score = await fetchDemoAssessment({
+    url: SUBMIT_URLS.home,
+    json_body: json_body,
+  });
+  useAssessment(score);
 }
 
-function useAssessment(score) {
-  // SEE: Code for handling the assessment goes here
-  showAssessmentInDemo(score);
+async function signup({token}) {
+  // Do not pass plain text credentials here. Always encrypt and follow security standards.
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      token: token,
+      username: "",
+      password: "",
+    },
+  });
+  // Fetch the demo assessment from backend.
+  const score = await fetchDemoAssessment({
+    url: SUBMIT_URLS.signup,
+    json_body: json_body,
+  });
+  useAssessment(score);
+}
+
+async function login({token}) {
+  // Do not pass plain text credentials here. Always encrypt and follow security standards.
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      token: token,
+      username: "",
+      password: "",
+    },
+  });
+  // Fetch the demo assessment from backend.
+  const score = await fetchDemoAssessment({
+    url: SUBMIT_URLS.login,
+    json_body: json_body,
+  });
+  useAssessment(score);
 }
 
-// ATTENTION: reCAPTCHA Example (Part 3) Ends
+async function store({token}) {
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      token: token,
+      items: {}.toString(),
+    },
+  });
+  // Fetch the demo assessment from backend.
+  const score = await fetchDemoAssessment({
+    url: SUBMIT_URLS.store,
+    json_body: json_body,
+  });
+  useAssessment(score);
+}
 
-function fetchDemoAssessment({ action, token }) {
-  const body = JSON.stringify({
+async function comment({token}) {
+  const json_body = JSON.stringify({
     recaptcha_cred: {
-      action,
-      token,
+      token: token,
+      comment: "",
```

Comment:
Shouldn't we pass the actual comment from the form?

Reply:
ditto